`azurerm_servicebus_subscription`: support client affine property, `azurerm_servicebus_subscription_rule`: add property `sql_filter_compatibility_level` (#17101)
hashicorp/terraform-provider-azurerm: website/docs/r/servicebus_subscription.html.markdown

* `status` - (Optional) The status of the Subscription. Possible values are `Active`,`ReceiveDisabled`, or `Disabled`. Defaults to `Active`.

* `client_scoped_subscription_enabled` - (Optional)  whether the subscription has an affinity to the client id. Defaults to `False`. 

> ~> **NOTE:** Client Scoped Subscription can only be used for JMS subscription.

Reviewer: what is a JMS subscription?

Author: Thanks Kt for the review, as the link that I put in the pr description, the full name of JMS subscription is Java message service (JMS) subscriptions: _While these are semantically similar to the [Subscriptions](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-queues-topics-subscriptions#topics-and-subscriptions) (that is, exist on a topic and enable publish/subscribe semantics), the Java Message Service spec introduces the concepts of Shared, Unshared, **Durable, and Non-durable attributes for a given subscription._

> * `client_id_for_client_scoped_subscription` - (Optional)  Specifies the Client ID of the application that created the client-scoped subscription.

Reviewer: is this client "scoped" or client affinity?
```suggestion
* `client_id` - (Optional)  The client id this subscription should have an affinity with.
```

the mixing of terms here is quite confusing

> * `client_scoped_subscription_enabled` - (Optional)  whether the subscription has an affinity to the client id. Defaults to `False`. 
> 
> ~> **NOTE:** Client Scoped Subscription can only be used for JMS subscription.
> 
> * `client_id_for_client_scoped_subscription` - (Optional)  Specifies the Client ID of the application that created the client-scoped subscription.
> 
> ~> **NOTE:** Client ID can be null or empty, but it must match the client ID set on the JMS client application. From the Azure Service Bus perspective, a null client ID and an empty client id have the same behavior. If the client ID is set to null or empty, it is only accessible to client applications whose client ID is also set to null or empty.
> 
> * `is_client_scoped_subscription_shareable` - (Optional) Whether the client scoped subscription is shareable. Defaults to `true`
> 
> * `is_client_scoped_subscription_durable` - (Optional) Whether the client scoped subscription is durable. This property can only be controlled from the application side.

Reviewer: would this make sense to wrap in a `client_affinity` block?
```suggestion
* `enabled` - (Optional)  whether the subscription has an affinity to a client id. Defaults to `False`. 

~> **NOTE:** Client Scoped Subscription can only be used for JMS subscription.

* `client_id` - (Optional)  Specifies the Client ID of the application that created the client-scoped subscription.

~> **NOTE:** Client ID can be null or empty, but it must match the client ID set on the JMS client application. From the Azure Service Bus perspective, a null client ID and an empty client id have the same behavior. If the client ID is set to null or empty, it is only accessible to client applications whose client ID is also set to null or empty.

* `shareable` - (Optional) Whether the client scoped subscription is shareable. Defaults to `true`

* `durable` - (Optional) Whether the client scoped subscription is durable. This property can only be controlled from the application side.
```

Author: Will put the related properties into a block.
